pkg/apis/core/validation/events.go

```diff
@@ -33,8 +38,88 @@ const (
 	NoteLengthLimit              = 1024
 )
 
-// ValidateEvent makes sure that the event makes sense.
-func ValidateEvent(event *core.Event) field.ErrorList {
+func ValidateEventCreate(event *core.Event, requestVersion schema.GroupVersion) field.ErrorList {
+	// Make sure events always pass legacy validation.
+	allErrs := legacyValidateEvent(event)
+	if requestVersion == core.SchemeGroupVersion || requestVersion == v1.SchemeGroupVersion || requestVersion == eventsv1beta1.SchemeGroupVersion {
```

Comment:
nit: instead of this if, wouldn't it be simpler to have:
```
if requestVersion != eventsv1.SchemeGroupVersion {
   ...
}
```

Same in validate update

Reply:
no, we want the stricter validation to apply to any future versions as well. we want to specifically call out core/v1 and events/v1beta1 as having looser validation

Reply:
ok - thanks for clarification

@liggitt - should we remove "core.SchemeGroupVersion" - requestVersion should always be external one, right?

Reply:
> should we remove "core.SchemeGroupVersion" - requestVersion should always be external one, right?

yes, that was my suggestion in https://github.com/kubernetes/kubernetes/pull/91645#discussion_r447338412

Reply:
I covered all SchemeGroupVersion under:
- "k8s.io/kubernetes/pkg/apis/core"
- "k8s.io/api/core/v1"
- "k8s.io/kubernetes/pkg/apis/events/v1beta1"

but `TestCreate` and `TestUpdate` in `/pkg/registry/core/event/storage:go_default_test` failed saying
```
Unexpected error: Event "foo1" is invalid: [eventTime: Required value, type: Invalid value: "": has invalid value: ]
```
`eventTime` should only be checked when it's `events/v1`, but I don't think those tests under `core/event/storage` will have `events/v1` as `requestVersion`. But it looks like it's neither of those three versions either. I'm not sure how to fix it..
Is there any difference between SchemeGroupVersion of "k8s.io/api/core/v1" and "k8s.io/kubernetes/pkg/apis/core/v1"?

Reply:
The unit tests in TestCreate/TestUpdate are not populating the context with the attributes a normal API request would have, so request version is coming in as an empty `schema.GroupVersion{}`. Your options are:

1. update the valid fixtures in TestCreate/TestUpdate to pass the stricter validation (I would recommend doing this)
2. relax validation for `schema.GroupVersion{}` as well

Reply:
Got it. I've update TestCreate/TestUpdate to match stricter validation.